src/vm/eventpipesession.cpp

```diff
@@ -89,4 +86,157 @@ EventPipeSessionProvider* EventPipeSession::GetSessionProvider(EventPipeProvider
+    EventPipeFile &fastSerializableObject,
+    EventPipeConfiguration &configuration,
+    LARGE_INTEGER stopTimeStamp)
+{
+    CONTRACTL
+    {
+        THROWS;
+        GC_NOTRIGGER;
+        MODE_ANY;
+        PRECONDITION(EventPipe::IsLockOwnedByCurrentThread());
+    }
+    CONTRACTL_END;
+
+    m_pBufferManager->WriteAllBuffersToFile(&fastSerializableObject, configuration, stopTimeStamp);
+}
+
+bool EventPipeSession::WriteEvent(
+        Thread *pThread,
+        EventPipeEvent &event,
+        EventPipeEventPayload &payload,
+        LPCGUID pActivityId,
+        LPCGUID pRelatedActivityId,
+        Thread *pEventThread,
+        StackContents *pStack)
+{
+    return m_pBufferManager->WriteEvent(pThread, *this, event, payload, pActivityId, pRelatedActivityId);
+}
+
+EventPipeEventInstance *EventPipeSession::GetNextEvent()
+{
+    CONTRACTL
+    {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+        PRECONDITION(!EventPipe::IsLockOwnedByCurrentThread());
+    }
+    CONTRACTL_END;
+
+    return m_pBufferManager->GetNextEvent();
+}
+
+#ifdef DEBUG
+    bool EventPipeSession::IsLockOwnedByCurrentThread()
+    {
+        return m_pBufferManager->IsLockOwnedByCurrentThread();
+    }
+#endif // DEBUG
+
+// void EventPipeSession::Enable(EventPipeProviderCallbackDataQueue *pEventPipeProviderCallbackDataQueue)
+// {
+//     CONTRACTL
+//     {
+//         THROWS;
+//         GC_TRIGGERS;
+//         MODE_ANY;
+//         // Lock must be held by EventPipe::Enable.
+//         PRECONDITION(EventPipe::IsLockOwnedByCurrentThread());
+//     }
+//     CONTRACTL_END;
+
+//     m_enabled = true;
+
+//     // The provider list should be non-NULL, but can be NULL on shutdown.
+//     if (m_pProviderList != NULL)
+//     {
+//         SListElem<EventPipeProvider *> *pElem = m_pProviderList->GetHead();
+//         while (pElem != NULL)
+//         {
+//             EventPipeProvider *pProvider = pElem->GetValue();
+
+//             // Enable the provider if it has been configured.
+//             EventPipeSessionProvider *pSessionProvider = GetSessionProvider(pProvider);
+//             if (pSessionProvider != NULL)
+//             {
+//                 EventPipeProviderCallbackData eventPipeProviderCallbackData =
+//                     pProvider->SetConfiguration(
+//                         true /* providerEnabled */,
+//                         pSessionProvider->GetKeywords(),
+//                         pSessionProvider->GetLevel(),
+//                         pSessionProvider->GetFilterData());
+//                 pEventPipeProviderCallbackDataQueue->Enqueue(&eventPipeProviderCallbackData);
+//             }
+
+//             pElem = m_pProviderList->GetNext(pElem);
+//         }
+//     }
+// }
+
+const EventPipeProviderConfiguration RundownProviders[] = {
+    {W("Microsoft-Windows-DotNETRuntime"), 0x80020138, static_cast<unsigned int>(EventPipeEventLevel::Verbose), NULL},       // Public provider.
```

Comment:
We should document what this magic number is and pull it out into a define. It looks like it is:
LoaderKeyword(0x8) | JitKeyword(0x10) | NgenKeyword (0x20) | unused_keyword (0x100) | JittedMethodILToNativeMapKeyword (0x20_000) | ThreadTransferKeyword (0x8000_0000)